lib/fetch-motif-doc.js

```diff
@@ -0,0 +1,145 @@
+      };
+    })
+  );
+
+  return _.sortBy(motifsList, ["targets"]);
+}
+
+/**
+ * Convert PWM file text into a matrix array of nucleotide frequency information.
+ * In the matrix, each row is an array of 4 elements long (one element is the frenquency value for each nucleotide) and there are n rows
+ * The PWM file text looks like this:
+ * >MA0630.1	SHOX
+ * A  [   179     74    852    991     15     98    416    301 ]
+ * C  [   301     30     48      2     15     50     48    187 ]
+ * G  [   162      9     55      5     27     84    424    379 ]
+ * T  [   357    886     45      2    943    768    111    133 ]
+ *
+ * @param {string} text PWM file text
+ * @returns an array of matrix
+ */
+function convertPwmTextToObj(text) {
+  // Split the file by row and create a new object
+  // The new object has an element corresponding to each row and each element is an array of the entries in that row, split by tab (the PWM file is tab-delineated)
+  const cells = text.split("\n").map((el) => el.split(/\s+/));
+  const matrix = [];
+  // Create a matrix without the index or rows that contain comments or unnecessary information and convert the frequency values from strings to numbers
+  cells.forEach((cell) => {
+    if (cell.length === 6) {
+      matrix.push([+cell[1], +cell[2], +cell[3], +cell[4]]);
+    }
+  });
+  const newMatrix = convert(matrix);
+  return newMatrix;
+}
+/**
+ * Convert jaspar file text into a matrix array of nucleotide frequency information
+ * The format of .jaspar files is approximately an inverted structure as .pwm files
+ * @param {string} text the jaspar file text
+ * @returns an array of matrix
+ */
+function convertJasparTextToObj(text) {
+  const cells = text.split("\n").map((el) => el.split(/\s+/));
+  let matrix = [];
+  const nucleotides = ["A", "C", "G", "T"];
+  for (let idx = 0; idx < cells.length; idx += 1) {
+    const cell = cells[idx];
+    // Finding first row of nuceotide information
+    if (nucleotides.includes(cell[0][0])) {
+      matrix = new Array(cell.length - 3);
+      for (let i = 0; i < matrix.length; i += 1) {
+        matrix[i] = new Array(4).fill(0);
+      }
+      break;
+    }
+  }
+  // Populating the arrays with the retrieved data
+  cells.forEach((cell) => {
+    if (nucleotides.includes(cell[0][0])) {
+      let rIdx = 0;
+      const nIdx = nucleotides.indexOf(cell[0][0]);
+      cell.forEach((c, cIdx) => {
+        if (c !== "]" && c !== "[" && cIdx > 0) {
+          matrix[rIdx][nIdx] = +c;
+          rIdx += 1;
+        }
+      });
+    }
+  });
+  const newMatrix = convert(matrix);
+  return newMatrix;
+}
+/**
+ * This function convert the values in pwm matrix so the values in each row sum to 1.
+ * @param {*} matrix matrix to convert
+ * @returns a cnverted matrix
+ */
+function convert(matrix) {
+  const newMatrix = matrix.map((cell) => {
+    const sum = cell.reduce((a, b) => +a + +b, 0);
+    const newCell = cell.map((element) => (element = +element / sum));
```

Comment:
A side effect inside a `.map()` is unusual. Could this just be `cell.map((element) => +element / sum)`?

Reply:
Thanks!